api/src/devcomp/domain/services/solution-service-qrocm-ind.js

```diff
@@ -37,66 +42,99 @@ function match({ answerValue, solution }) {
 }
 
 function _applyTreatmentsToSolutions(solutions, enabledTreatments, qrocBlocksTypes = {}) {
-  return _.forEach(solutions, (solution, solutionKey) => {
-    solution.forEach((variant, variantIndex) => {
-      if (qrocBlocksTypes[solutionKey] === 'select') {
-        solutions[solutionKey][variantIndex] = applyTreatments(variant, []);
-      } else {
-        solutions[solutionKey][variantIndex] = applyTreatments(variant, enabledTreatments);
+  const treatedSolutions = {};
+  for (const solutionKey in solutions) {
+    const solutionVariants = solutions[solutionKey];
+    const solutionType = qrocBlocksTypes[solutionKey];
+
+    treatedSolutions[solutionKey] = solutionVariants.map((variant) => {
+      switch (solutionType) {
+        case 'select':
+          return applyTreatments(variant, []);
+        case 'text':
+        default:
+          return applyTreatments(variant, enabledTreatments);
       }
     });
-  });
+  }
+
+  return treatedSolutions;
 }
 
 function _applyTreatmentsToAnswers(answers, enabledTreatments, qrocBlocksTypes = {}) {
-  return _.forEach(answers, (answer, answerKey) => {
-    if (qrocBlocksTypes[answerKey] === 'select') {
-      answers[answerKey] = applyTreatments(answer, []);
-    } else {
-      answers[answerKey] = applyTreatments(answer, enabledTreatments);
+  const treatedAnswers = {};
+  for (const answerKey in answers) {
+    const answer = answers[answerKey];
+    const answerType = qrocBlocksTypes[answerKey];
+    switch (answerType) {
+      case 'select':
+        treatedAnswers[answerKey] = applyTreatments(answer, []);
+        break;
+      case 'text':
+      default:
+        treatedAnswers[answerKey] = applyTreatments(answer, enabledTreatments);
+        break;
     }
-  });
+  }
+
+  return treatedAnswers;
 }
 
 function _areApproximatelyEqualAccordingToLevenshteinDistanceRatio(answer, solutionVariants) {
   let smallestLevenshteinDistance = answer.length;
-  solutionVariants.forEach((variant) => {
+  for (const variant of solutionVariants) {
```

Comment:
Question - ici ce changement est juste pour etre consistent avec les autres et on utilise les `for` au lieu de `forEach`?

Reply:
J'ai un peu insisté sur cette modification. On est dans du code complexe, au cœur du contexte métier de validation de Pix. Ici je pense que le code doit être le plus simple possible pour que ce ne soit pas douloureux de revenir dessus quand on aura besoin. 

Je pense notamment à https://github.com/zakirullin/cognitive-load?tab=readme-ov-file#feature-rich-languages

Reply:
super lien merci @yannbertrand 👍